litestar/cli/commands/core.py

```diff
@@ -59,12 +61,19 @@ def info_command(app: Litestar) -> None:
 )
 @option("--host", help="Server under this host", default="127.0.0.1", show_default=True)
 @option("--debug", help="Run app in debug mode", is_flag=True)
+@option(
+    "--app-dir",
+    help="Look for APP in the specified directory, by adding this to the PYTHONPATH. Defaults to the current working directory.",
+    default=None,
+    show_default=False,
+)
```

Comment:
This should go on the root command. If we only inject this into uvicorn it won't work, since the autodiscovery logic will still only look in the CWD, and therefore not find the app in the first place. If it's on the root command that means it'll also work for commands besides `run`.

Maybe we can just set the `PYTHONPATH` env-var before the autodiscovery process starts?

Reply:
I've done exactly this.  It should now work with all of the commands.  The syntax looks something like:
```shell
litestar --app-dir /home/oracle/Code/Starlite/litestar-fullstack/src/ --app app.asgi:create_app  info
```

Reply:
Maybe we could also think about supporting `src/` by default, since it's relatively common? I'm thinking it could be as simple as detecting if it's there, and if no other ASGI app is found try again but set this as the `PYTHONPATH`. Wdyt?